neqo-transport/src/connection/mod.rs

```diff
@@ -1065,9 +1065,17 @@ impl Connection {
             let pto = rtt.pto(self.confirmed());
 
             let idle_time = self.idle_timeout.expiry(now, pto);
-            qtrace!([self], "Idle/keepalive timer {:?}", idle_time);
+            qtrace!([self], "Idle timer {:?}", idle_time);
             delays.push(idle_time);
 
+            if self.streams.need_keep_alive() {
+                if let Some(keep_alive_time) = self.idle_timeout.maybe_keep_alive_timeout(now, pto)
+                {
+                    qtrace!([self], "Keep alive timer {:?}", keep_alive_time);
+                    delays.push(keep_alive_time);
```

Comment:
When you add another delay to the list, it might be wise to increase the size of the collection.  We're using a `SmallVec<[_; 6]>` right now, but we might need that to be 7 now.